### PR TITLE
Reset game

### PR DESCRIPTION
**Pull request by rmrose78**
# Description
- Die selected will have the select status removed if the die/dice did not score points when the roll button is pressed.
- If all six dice through one roll or multiple rolls in a turn lead to points, reset dice so all dice can be rolled again to score more points.
- Reset game button added.
- Cleaned up code and added the fixes from prior PR.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Dice reset on roll: Select the dice that score points and select dice that do not lead to points. Press the 'Roll' button and the dice that did not score points will reset.
- Dice reset if all six score: Roll dice until all six dice score points. Press the 'Roll' button and all dice should roll and be selectable. Also, the score for the round should not be reset.
- Game reset: Play the game and hit the 'Reset' button all scores should zero out, dice are changed to blank images, and player 1 selected 

- [x] Dice reset on roll
- [x] Dice reset if all six score
- [x] Game reset

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -223,12 +218,93 @@ points = {
...
+    dice.positionValues = [0, 0, 0, 0, 0, 0];
+    dice.shouldRoll = [true, true, true, true, true, true];
+    scoring.tempShouldRoll = [true, true, true, true, true, true];
+    dice.display();
+  }
+});
+
+// --- RESET --- \\
+
+resetBtn.addEventListener("click", () => {
```
Comment (axecopfire) [inline main.js:295]: If you put this inside a function called `reset` you wouldn't need the comment

**Review by axecopfire — APPROVED** [review]

File: main.js [hunk 2]
```diff
@@ -59,162 +47,169 @@ const dice = {
...
+    this.readDiceCombos(freqObj);
   },
 
   diceSelected: function () {
-    diceSelected = [];
+    diceSelectedArr = [];
     for (const die of dice.positions) {
       if (
-        dicePositions[die].classList[1] === "select" &&
+        dicePositions[die].classList.contains("select") &&
```
Comment (axecopfire) [inline main.js:118]: Shouldn't it be `includes`? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Comment (rmrose78) [reply main.js:118]: I tried includes() and it didn't work. Googlized it and found that includes() works for arrays, but for nodes contains() is needed.

Comment (rmrose78) [reply main.js:118]: <code>includes</code> *testing to see if this does the highlight thing on code.

Comment (rmrose78) [reply main.js:118]: Sweet

**Pull request merged**